### DISCUSSION TITLE
Aggregate over path elements

### DISCUSSION THREAD

**Discussion by rrozanka** [plausible/analytics#3065] [Anything Else]
Hello! I'm running a website that has a URL format of:
```
/sites/<site1>/<page1>
/sites/<site1>/<page2>
/sites/<site1>/<page3>

...

/sites/<site2>/<page1>
```

Out-of-the-box we get page-level { page1, page2, ..., pageN } analytics, however, our client is interested in seeing the roll-up/aggregate statistics on the site-level { site1, site2, ..., siteN}. While for # of page views, I can easily just loop through the pages and sum it up, obviously this is not the case for the unique visitors.

So far I tried the following:
* send a custom property called "page" with the pageview event: `{ page: <page1> }`. I can then use the filter can see the correct analytics for the site-level (including unique visitors, nice!). This solution "seems" like a hack, because I won't have `/sites/<site1>` on the dashboard nor in the exported data
* I tried to switch to the manual pageview event triggering. My plan was to "simply" trigger pageview twice: 1) for the site-level URL, 2) for the page-level URL. I ended up with seeing `/sites/<site1>` on the dashboard and in the export, unique visitors are fine, yet # of pageviews are inflated (due to the 2x pageview event sent), when using a filter over the page: `/sites/<site1>**`.

Ideally, we'd like to see a report like this:
```
/sites/<site1> [unique visitors: 1 - 9 (a correct # will be provided by Plausible, right?), pageviews: 5 + 10 + 15 = 30]
/sites/<site1>/<page1> [unique visitors: 1, pageviews: 5]
/sites/<site1>/<page2> [unique visitors: 3, pageviews: 10]
/sites/<site1>/<page3> [unique visitors: 5, pageviews: 15]
```

Thanks for any help!

**Comment by metmarkosaric** [comment]
hi  @rrozanka! without the actual domain and dashboard it's difficult to understand your usecase, but we do have the roll-up reporting: https://plausible.io/docs/plausible-script#is-there-a-roll-up-view. it allows you to aggregate stats from multiple sites and see them all together in one dashboard while still keeping the individual site stats on their own separate dashboards

**Reply by rrozanka** [reply]
Morning @metmarkosaric! Our use case is a set of microsites, all hosted under a single domain, like so: `<domain.com>/sites/<client>/<page>`. We have a single Plausible site, so a single dashboard.  Our clients asked us whether it is possible to have aggregate stats on the `<client>` level + a per-`<page>` stats as well.

I just realized that we can use a filter for that: `Page is /sites/client1**` which will aggregate over `client1` pages, does it make sense? If that's true, then the last problem we are left with is how to see the per-client aggregate in the exported CSV.

Hence the following example. All visits were done by me, from the different browsers (to act as 2 unique users):
```
/sites/plausible1/A - 1 unique visitor { visitor1 }, 2 pageviews
/sites/plausible1/B - 2 unique visitors { visitor1, visitor2 }, 3 pageviews
/sites/plausible1/C - 2 unique visitors { visitor1, visitor2 }, 2 pageviews
```

<img width="468" alt="image" src="https://github.com/plausible/analytics/assets/4865339/0f92f7b8-0c45-4a3a-9b61-e669167b85c0">

Ideally, we would like this report to include an aggregate, like so:
<img width="460" alt="image" src="https://github.com/plausible/analytics/assets/4865339/84e52f46-258f-4e43-9bf8-f0b066218116">

The dashboard gives me a correct aggregate:
<img width="708" alt="image" src="https://github.com/plausible/analytics/assets/4865339/36ae7e50-ac22-4bd7-b226-19b2a4e9b852">

Is this something that can be achieved, or we need to build something custom on our end to support that?

**Reply by metmarkosaric** [reply]
yes, that's possible too. there's also the "contains" option for pages in the "filter" menu which does the same as wildcards. and if you click on the csv export button when the dashboard is filtered, the export will only include the filtered data. you can read more about filtering here https://plausible.io/docs/filters-segments

**Reply by rrozanka** [reply]
yes, that's amazing, however what I'm missing in the exported data is this top-level aggregate numbers, and I'm not sure whether that's something possible or not (ptal updated comment above)

**Reply by metmarkosaric** [reply]
i see. there's the visitors.csv file in the export which should have that data but it would be split according to the chart interval rather than be aggregate

**Reply by rrozanka** [reply]
right. ok, so I guess the way to go for us would be to build our custom CSV export, which will use Plausible API and do what we want (because that's possible for sure). Not the end of the world. FYI we are moving away from GA4/GTM combo (because of the privacy issues) and we were unable to do the same over there as well. Thanks for your time!
